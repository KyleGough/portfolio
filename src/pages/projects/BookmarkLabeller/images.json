[
  { "src": "/img/bookmark-labeller.jpg", "caption": "Bookmark Labeller Extension Popup" },
  { "src": "/img/bookmark-labeller-folder.jpg", "caption": "Labelled bookmark folder" }
]
