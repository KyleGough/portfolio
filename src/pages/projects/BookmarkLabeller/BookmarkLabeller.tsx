import React, { useEffect } from 'react';

import { Divider } from '../../../components/Divider';
import { ImageFigure } from '../../../components/ImageFigure';
import { Pagination } from '../../../components/Pagination';
import { ProjectHeader } from '../../../components/ProjectHeader';
import { Screenshots } from '../../../components/Screenshots';
import { Section } from '../../../components/Section';
import { getProjectData } from '../../../data';
import images from './images.json';

const project = getProjectData('bookmark-labeller');

export const BookmarkLabeller: React.FC = () => {
  useEffect(() => {
    document.title = 'Bookmark Labeller Extension';
  }, []);

  return (
    <>
      <ProjectHeader project={project} />

      <Divider />

      <ImageFigure image={images[0]} />

      <Divider />

      <Section>
        <h2 className="project-header">Inspiration</h2>
        <p className="max-w-reading">
          I created this browser extension for two reasons: I often found myself
          labelling bookmarks for work such as marking PRs as merged, open, or
          closed and wanted a simpler way to achieve this. Secondly, I wanted to
          learn and experiment with the WebExtensions API. My hope is that
          people other than myself can find this extension useful.
        </p>
      </Section>

      <Divider />

      <Section>
        <h2 className="project-header">Features</h2>
        <ul className="project-list">
          <li>
            Lightweight and minimalistic &#40;<strong>13.2 KB</strong>&#41;
          </li>
          <li>Quickly label current bookmark from the extension popup</li>
          <li>Label bookmarks and folders with the context menu</li>
          <li>Customise favourite emojis</li>
        </ul>
      </Section>

      <Divider />

      <Screenshots images={images} />

      <Divider />

      <Pagination
        previousTitle="Portfolio"
        previousLink="/projects/portfolio"
      />
    </>
  );
};
