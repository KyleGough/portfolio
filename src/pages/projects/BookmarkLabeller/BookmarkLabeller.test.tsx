import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { BookmarkLabeller } from './BookmarkLabeller';

describe('BookmarkLabeller component', () => {
  it('renders', async () => {
    const { container } = render(
      <BrowserRouter>
        <BookmarkLabeller />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
