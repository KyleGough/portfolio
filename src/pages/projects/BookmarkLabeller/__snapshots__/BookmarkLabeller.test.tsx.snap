// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkLabeller component renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h1
      class="project-title"
    >
      Bookmark Labeller
    </h1>
    <p
      class="text-link-hover my-4"
    >
      <time
        datetime="2022-10"
      >
        October 2022
      </time>
    </p>
    <p
      class="mb-4 max-w-reading"
    >
      Bookmark Labeller is a lightweight and minimalistic Firefox browser extension that allows you to label bookmarks with your favourite emojis. The extension comes with 9 default emojis which can be customised. Use the action button to select emojis or right-click bookmarks to add emojis with the context menu.
    </p>
    <ul
      class="project-list"
    >
      <li
        class="mb-4"
      >
        <span>
          Explore this project on 
        </span>
        <a
          class="text-link hover:text-link-hover focus:text-link-hover"
          href="https://github.com/KyleGough/bookmark-labeller-extension"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub
        </a>
      </li>
    </ul>
    <div
      class="flex flex-row flex-wrap items-center mt-8 gap-4"
    >
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        JavaScript
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        WebExtensions API
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        HTML
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        CSS
      </div>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <img
      alt="Bookmark Labeller Extension Popup"
      class="border-2 border-white rounded-lg text-center items-center shadow mx-auto"
      src="/img/bookmark-labeller.jpg"
    />
    <p
      class="mt-4 mb-2 text-sm text-center text-light font-thin"
    >
      Bookmark Labeller Extension Popup
    </p>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <div
      class="flex"
    >
      <img
        alt="Firefox Logo"
        class="w-8 h-8 mr-2"
        src="firefox.svg"
      />
      <a
        class="text-link hover:text-link-hover focus:text-link-hover"
        href="https://addons.mozilla.org/en-US/firefox/addon/bookmark-labeller/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <p
          class="text-2xl"
        >
          Install from the Mozilla Add-On Marketplace
        </p>
      </a>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="project-header"
    >
      Inspiration
    </h2>
    <p
      class="max-w-reading"
    >
      I created this browser extension for two reasons: I often found myself labelling bookmarks for work such as marking PRs as merged, open, or closed and wanted a simpler way to achieve this. Secondly, I wanted to learn and experiment with the WebExtensions API. My hope is that people other than myself can find this extension useful.
    </p>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="project-header"
    >
      Features
    </h2>
    <ul
      class="project-list"
    >
      <li>
        Lightweight and minimalistic
      </li>
      <li>
        Quickly label current bookmark from the extension popup
      </li>
      <li>
        Label bookmarks and folders with the context menu
      </li>
      <li>
        Customise favourite emojis
      </li>
    </ul>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="text-3xl mb-12"
    >
      Screenshots
    </h2>
    <div
      class="grid grid-cols-12 gap-4"
    >
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Bookmark Labeller Extension Popup"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/bookmark-labeller.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Bookmark Labeller Extension Popup
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Labelled bookmark folder"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/bookmark-labeller-folder.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Labelled bookmark folder
        </figcaption>
      </figure>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <div
      class="flex justify-between"
    >
      <a
        class="flex items-center fill-link text-link hover:fill-link-hover hover:text-link-hover"
        href="/projects/portfolio"
      >
        <svg
          class="w-6 h-6"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
            opacity=".87"
          />
          <path
            d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
          />
        </svg>
        <div
          class="text-center ml-4"
        >
          <p
            class="text-sm"
          >
            Previous Project
          </p>
          <p>
            Portfolio
          </p>
        </div>
      </a>
      <a
        class="flex items-center fill-disabled text-disabled cursor-not-allowed"
        href="/"
      >
        <div
          class="text-center mr-4"
        >
          <p
            class="text-sm"
          >
            Next Project
          </p>
          <p>
            -
          </p>
        </div>
        <svg
          class="w-6 h-6"
          enable-background="new 0 0 24 24"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0,0h24v24H0V0z"
              fill="none"
            />
          </g>
          <g>
            <polygon
              points="6.23,20.23 8,22 18,12 8,2 6.23,3.77 14.46,12"
            />
          </g>
        </svg>
      </a>
    </div>
  </div>
</div>
`;
